qiskit/result/utils.py

```diff
@@ -35,8 +41,8 @@ def marginal_counts(result, indices=None):
     Raises:
         QiskitError: in case of invalid indices to marginalize over.
     """
-    from qiskit.result.result import Result
-    from qiskit.result.postprocess import _bin_to_hex
```

Comment:
Why moving these imports out?

Reply:
I saw other method codeblocks in Qiskit and all of them followed the convention of Global imports. Which is why I moved the imports out of the method body to follow with the same convention.

Reply:
Sorry. I meant the true-branch. In the case the argument is a dictionary, the behavior is `inplace=False` since `_marginalize` does not modify it. I tested with 8cddc469e171aabd191510f97b66fc12e4e1630f, have a look. I'm passing your test. I might be wrong, can you show me a counter example and add it to the tests?

Reply:
Yes you are right, it does not affect if the result argument if it is a dictionary. However, in that case, there is also no need for ```.get_counts(0)```. I am creating a test using the dictionary then.

Reply:
Added Tests for testing ```marginal_counts()``` with a dictionary and see if the dict object gets changed or not. Please have a look at commit [afda63a](https://github.com/Qiskit/qiskit-terra/pull/4929/commits/afda63aa55a13cc10fdfb9a4f0228b48bbf3b3f3) . Is this okay?